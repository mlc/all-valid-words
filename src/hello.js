import { successResponse, runWarm } from './utils';

const hello = (event, context, callback) => {
  // successResponse handles wrapping the response in an API Gateway friendly
  // format (see other responses, including CORS, in `./utils/lambda-response.js)
  const response = successResponse({
    message: 'Go Serverless v1.0! Your function executed successfully!',
    input: event,
  });

  callback(null, response);

  // Use this code if you don't use the http event with the LAMBDA-PROXY integration
  // callback(null, { message: 'Go Serverless v1.0! Your function executed successfully!', event });
};

// runWarm function handles pings from the scheduler so you don't
// have to put that boilerplate in your function.
export default runWarm(hello);
