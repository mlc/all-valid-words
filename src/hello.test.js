import hello from './hello';

describe('hello', () => {
  it('executes as expected', () => {
    const cb = jest.fn();
    hello({}, {}, cb);
    expect(cb).toBeCalled();
    expect(cb).toMatchSnapshot();
  });
});
