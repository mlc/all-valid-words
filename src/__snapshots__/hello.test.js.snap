// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hello executes as expected 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      null,
      Object {
        "body": "{\\"message\\":\\"Go Serverless v1.0! Your function executed successfully!\\",\\"input\\":{}}",
        "statusCode": 200,
      },
    ],
  ],
}
`;
